lib/core/sdam/server.js

```diff
@@ -148,16 +148,13 @@ class Server extends EventEmitter {
       { bson: this.s.bson }
     );
 
-    // NOTE: this should only be the case if we are connecting to a single server
-    poolOptions.reconnect = true;
+    // NOTE: reconnect is explicitly false because of the server selection loop
+    poolOptions.reconnect = false;
```

Comment:
side-note: we currently have a ticket about the buckets of SDAM options going away when we do 4.0. This is already one of them.